### PR TITLE
Wednesday edits

### PR DESCRIPTION
**Pull request by mrose17**
found that if the network was *really* slow, `run` might get invoked before the wallet got created.

auditor: @diracdeltas

### CODE REVIEW

File: app/ledger.js [hunk 1]
```diff
@@ -867,7 +867,11 @@ var run = (delayTime) => {
   }
 
   if (delayTime === 0) {
-    delayTime = client.timeUntilReconcile()
+    try {
+      delayTime = client.timeUntilReconcile()
+    } catch (ex) {
+      delayTime = random.randomInt({ min: 1, max: 10 * msecs.minute })
```
Comment (diracdeltas) [inline app/ledger.js]: if this random delay happens to be small enough, is it still possible for `run` to be invoked before wallet initialization? i'm not understanding how this change fixes the problem stated in the PR

**Comment by mrose17** [comment]
The patch catches the error that may be thrown and then endures a non-zero wait before the next call to sync, which ultimately results in a call to run

**Comment by diracdeltas** [comment]
> endures a non-zero wait before the next call to sync, which ultimately results in a call to run

say that the wait is 1ms. will that cause a race condition problem given slow network speeds?

**Comment by mrose17** [comment]
Race condition? No.  Inefficient? Yes.

Tell me the lower bound that you like... Thanks!

**Comment by diracdeltas** [comment]
this is fine as long as it's not going to throw another error. for future reference, please put a longer explanation in the commit message itself (so someone looking through git log can understand the change without having to reference PRs).

**Comment by mrose17** [comment]
+1

**Comment by diracdeltas** [comment]
cherry-picked

**Pull request closed without merging**